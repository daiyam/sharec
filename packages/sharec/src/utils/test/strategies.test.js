const path = require('path')
const { readFileSync } = require('fs')
const { withYaml } = require('../strategies')

const yamlFixture01 = readFileSync(
  path.resolve(__dirname, './fixtures/withYaml_01.yml'),
  'utf8',
)
const yamlFixture02 = readFileSync(
  path.resolve(__dirname, './fixtures/withYaml_02.yml'),
  'utf8',
)
const yamlFixture03 = readFileSync(
  path.resolve(__dirname, './fixtures/withYaml_03.yml'),
  'utf8',
)

describe('strategies', () => {
  describe('withYaml', () => {
    const strategy = (a, b) => ({
      ...a,
      ...b,
    })

    it('should process yaml input with strategy', () => {
      const yamledStrategy = withYaml(strategy)

      expect(yamledStrategy(yamlFixture01, yamlFixture02)).toEqual(
        yamlFixture03,
      )
    })
  })
})
