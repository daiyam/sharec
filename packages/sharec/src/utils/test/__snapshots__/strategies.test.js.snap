// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`strategies withYaml should process yaml input with strategy 1`] = `
"env:
  browser: true
  es6: true
  node: true
  jest: true
globals:
  foo: true
  bar: true
extends: airbnb
rules:
  space-before-function-paren: 1
  no-console: 0
  max-len:
    - warn
    - 120
    - 4
    - ignoreUrls: false
      ignoreComments: true
      ignoreStrings: false
parserOptions:
  parser: babel-eslint
  ecmaVersion: 7
  ecmaFeatures:
    spread: true
  sourceType: module
"
`;
